cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -215,17 +228,54 @@ impl NetworkRunnable for Cmd {
             return Err(Error::OnlyEd25519AccountsAllowed);
         };
 
-        let key = stellar_strkey::ed25519::PublicKey(bytes.into());
+        let source_account = stellar_strkey::ed25519::PublicKey(bytes.into());
+        let contract_id_preimage = ContractIdPreimage::Address(ContractIdPreimageFromAddress {
+            address: ScAddress::Account(AccountId(PublicKey::PublicKeyTypeEd25519(
+                source_account.0.into(),
+            ))),
+            salt: Uint256(salt),
+        });
+        let contract_id =
+            get_contract_id(contract_id_preimage.clone(), &network.network_passphrase)?;
+        let raw_wasm = if let Some(wasm) = self.wasm.as_ref() {
+            wasm::Args { wasm: wasm.clone() }.read()?
+        } else {
+            get_remote_wasm_from_hash(&client, &wasm_hash).await?
+        };
+        let entries = soroban_spec_tools::contract::Spec::new(&raw_wasm)?.spec;
+        let res = soroban_spec_tools::Spec::new(entries.clone());
+        let constructor_params = if let Ok(func) = res.find_function(CONSTRUCTOR_FUNCTION_NAME) {
+            if func.inputs.len() == 0 {
+                None
+            } else {
+                let mut slop = vec![OsString::from(CONSTRUCTOR_FUNCTION_NAME)];
+                slop.extend_from_slice(&self.slop);
+                Some(
+                    arg_parsing::build_host_function_parameters(
+                        &stellar_strkey::Contract(contract_id.0),
+                        &slop,
+                        &entries,
+                        config,
+                    )?
+                    .2,
+                )
+            }
+        } else {
+            None
+        };
+
         // Get the account sequence number
-        let account_details = client.get_account(&key.to_string()).await?;
+        let account_details = client
+            .get_account(&config.source_account()?.to_string())
```

Comment:
Why did this change to accessing config.source_account(), when higher up the source account is defined as coming from the bytes? Iiuc both would need to be the same, what's the reason to change this location?